subworkflows/local/shortread_adapterremoval.nf

```diff
@@ -0,0 +1,89 @@
+/*
+Process short raw reads with AdapterRemoval
+*/
+
+include { ADAPTERREMOVAL as ADAPTERREMOVAL_SINGLE       } from '../../modules/nf-core/modules/adapterremoval/main'
+include { ADAPTERREMOVAL as ADAPTERREMOVAL_PAIRED       } from '../../modules/nf-core/modules/adapterremoval/main'
+include { CAT_FASTQ                                     } from '../../modules/nf-core/modules/cat/fastq/main'
+
+workflow SHORTREAD_ADAPTERREMOVAL {
+
+    take:
+    reads // [[meta], [reads]]
+
+    main:
+    ch_versions = Channel.empty()
+    ch_multiqc_files      = Channel.empty()
+
+    ch_input_for_adapterremoval = reads
+                                    .branch{
+                                        single: it[0]['single_end'] == true
+                                        paired: it[0]['single_end'] == false
+                                    }
+
+    ADAPTERREMOVAL_SINGLE ( ch_input_for_adapterremoval.single, [] )
+    ADAPTERREMOVAL_PAIRED ( ch_input_for_adapterremoval.paired, [] )
+
+    if ( params.shortread_clipmerge_mergepairs && !params.shortread_clipmerge_excludeunmerged ) {
+        ch_adapterremoval_for_cat = ADAPTERREMOVAL_PAIRED.out.collapsed
+                                                .mix(
+                                                    ADAPTERREMOVAL_PAIRED.out.collapsed_truncated,
+                                                    ADAPTERREMOVAL_PAIRED.out.singles_truncated,
+                                                    ADAPTERREMOVAL_PAIRED.out.pair1_truncated,
+                                                    ADAPTERREMOVAL_PAIRED.out.pair2_truncated
+                                                    )
+                                                .map {
+                                                    meta, reads ->
+                                                        def meta_new = meta.clone()
+                                                        meta_new['single_end'] = true
+
+                                                        [ meta_new, reads ]
+                                                    }
+                                                    .groupTuple(by: 0)
+
+
+        ch_adapterremoval_reads_prepped_pe = CAT_FASTQ ( ch_adapterremoval_for_cat ).reads
+
+        ch_adapterremoval_reads_prepped = ch_adapterremoval_reads_prepped_pe.mix( ADAPTERREMOVAL_SINGLE.out.singles_truncated )
```

Comment:
Since the first channel is not used any further, maybe combine these two (formatting to be adjusted).

```suggestion
        ch_adapterremoval_reads_prepped = CAT_FASTQ ( ch_adapterremoval_for_cat ).reads.mix( ADAPTERREMOVAL_SINGLE.out.singles_truncated )
```